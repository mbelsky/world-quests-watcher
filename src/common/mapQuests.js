import { regions } from "./const.js";

export function mapQuestsByRegion(quests) {
  const map = regions.reduce((result, region) => {
    result[region] = {};
    return result;
  }, {});

  quests.forEach((quest) => {
    const { id, region } = quest;

    if (region in map) {
      map[region][id] = quest;
    } else {
      console.warn(`Unexpected region value`, quest);
    }
  });

  return map;
}
