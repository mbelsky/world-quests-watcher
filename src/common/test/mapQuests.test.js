import { mapQuestsByRegion } from "../mapQuests.js";

const quests = [
  {
    endsAt: "11 hours",
    href: "/quest=52986",
    id: "52986",
    region: "na",
  },
  {
    endsAt: "11 hours",
    href: "/quest=52989",
    id: "52989",
    region: "na",
  },
  {
    endsAt: "11 hours",
    href: "/quest=52981",
    id: "52981",
    region: "na",
  },
  {
    endsAt: "11 hours",
    href: "/quest=62989",
    id: "62989",
    region: "na",
  },
  {
    endsAt: "11 hours",
    href: "/quest=62989",
    id: "62989",
    region: "eu",
  },
  {
    endsAt: "11 hours",
    href: "/quest=62981",
    id: "62981",
    region: "eu",
  },
  {
    endsAt: "11 hours",
    href: "/quest=52989",
    id: "52989",
    region: "eu",
  },
];

test("mapQuestsByRegion", () => {
  expect(mapQuestsByRegion(quests)).toMatchSnapshot();
});

test("with unexpected region", () => {
  expect(
    mapQuestsByRegion([
      {
        id: "52986",
        region: "kr",
      },
    ])
  ).toEqual({ eu: {}, na: {} });
});
