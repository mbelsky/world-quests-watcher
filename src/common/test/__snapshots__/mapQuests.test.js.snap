// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapQuestsByRegion 1`] = `
Object {
  "eu": Object {
    "52989": Object {
      "endsAt": "11 hours",
      "href": "/quest=52989",
      "id": "52989",
      "region": "eu",
    },
    "62981": Object {
      "endsAt": "11 hours",
      "href": "/quest=62981",
      "id": "62981",
      "region": "eu",
    },
    "62989": Object {
      "endsAt": "11 hours",
      "href": "/quest=62989",
      "id": "62989",
      "region": "eu",
    },
  },
  "na": Object {
    "52981": Object {
      "endsAt": "11 hours",
      "href": "/quest=52981",
      "id": "52981",
      "region": "na",
    },
    "52986": Object {
      "endsAt": "11 hours",
      "href": "/quest=52986",
      "id": "52986",
      "region": "na",
    },
    "52989": Object {
      "endsAt": "11 hours",
      "href": "/quest=52989",
      "id": "52989",
      "region": "na",
    },
    "62989": Object {
      "endsAt": "11 hours",
      "href": "/quest=62989",
      "id": "62989",
      "region": "na",
    },
  },
}
`;
